sklearn/preprocessing/_function_transformer.py

```diff
@@ -1,20 +1,34 @@
 import warnings
+from numbers import Number
 
 import numpy as np
 
 from ..base import BaseEstimator, TransformerMixin, _fit_context
 from ..utils._param_validation import StrOptions
-from ..utils._set_output import _get_output_config
+from ..utils._set_output import ADAPTERS_MANAGER, _get_output_config
 from ..utils.metaestimators import available_if
 from ..utils.validation import (
     _allclose_dense_sparse,
     _check_feature_names_in,
+    _get_feature_names,
     _is_pandas_df,
     _is_polars_df,
     check_array,
 )
 
 
+def _is_registered_adapter(*, adapter_name):
+    return adapter_name in ADAPTERS_MANAGER.adapters
+
+
+def _get_adapter_from_container(container):
+    module_name = container.__module__.split(".")[0]
+    try:
+        return ADAPTERS_MANAGER.adapters[module_name]
+    except KeyError:
+        return None
```

Comment:
what's the consequence here? Probably failing at a later stage? Wouldn't it be better to fail here explicitly with a message that the user can understand and know their dataframe is not supported?

Reply:
Here, I think that we can reraise indeed. I took some code from `_set_output` that leverage the `None` value which is not the case here.